rcl/test/rcl/test_graph.cpp

```diff
@@ -310,7 +313,218 @@ check_graph_state(
+  rcl_ret_t ret;
+  const rcl_guard_condition_t *graph_guard_condition =
+    rcl_node_get_graph_guard_condition(this->node_ptr);
+
+  const rcl_guard_condition_t *remote_graph_guard_condition =
+    rcl_node_get_graph_guard_condition(this->node_ptr);
+
+  auto remote_node_ptr = new rcl_node_t;
+  *remote_node_ptr = rcl_get_zero_initialized_node();
+  const char *remote_node_name = "remote_graph_node";
+  rcl_node_options_t node_options = rcl_node_get_default_options();
+  ret = rcl_node_init(remote_node_ptr, remote_node_name, "", &node_options);
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+
+  // Now create a publisher on "topic_name" and check that it is seen.
+  rcl_publisher_t pub = rcl_get_zero_initialized_publisher();
+  rcl_publisher_options_t pub_ops = rcl_publisher_get_default_options();
+  auto ts = ROSIDL_GET_MSG_TYPE_SUPPORT(test_msgs, msg, Primitives);
+  ret = rcl_publisher_init(&pub, this->node_ptr, ts, topic_name.c_str(), &pub_ops);
+  EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Now create two subscribers.
+  rcl_subscription_t sub = rcl_get_zero_initialized_subscription();
+  rcl_subscription_options_t sub_ops = rcl_subscription_get_default_options();
+  ret = rcl_subscription_init(&sub, this->node_ptr, ts, topic_name.c_str(), &sub_ops);
+  EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Now create a subscriber.
+  rcl_subscription_t sub2 = rcl_get_zero_initialized_subscription();
+  rcl_subscription_options_t sub_ops2 = rcl_subscription_get_default_options();
+  ret = rcl_subscription_init(&sub2, remote_node_ptr, ts, topic_name.c_str(), &sub_ops2);
+  EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  const char * service_name = "test_service";
+  rcl_service_t service = rcl_get_zero_initialized_service();
+  rcl_service_options_t service_options = rcl_service_get_default_options();
+  auto ts1 = ROSIDL_GET_SRV_TYPE_SUPPORT(test_msgs, Primitives);
+  ret = rcl_service_init(&service, this->node_ptr, ts1, service_name, &service_options);
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  size_t number_of_tries = 20;
+  rcutils_string_array_t node_names = rcutils_get_zero_initialized_string_array();
+  rcutils_string_array_t node_namespaces = rcutils_get_zero_initialized_string_array();
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT({
+                                      ret = rcutils_string_array_fini(&node_names);
+                                      ASSERT_EQ(RCUTILS_RET_OK, ret);
+                                      ret = rcutils_string_array_fini(&node_namespaces);
+                                      ASSERT_EQ(RCUTILS_RET_OK, ret);
+                                  });
+  size_t attempts = 0;
+  size_t max_attempts = 4;
+  while (node_names.size < 3) {
```

Comment:
I think I know the purpose of this loop, but comment might be nice.

Reply:
Adding comment